src/Integration/Binding/CSharpVBBindingConfig.cs

```diff
@@ -40,9 +52,17 @@ public CSharpVBBindingConfig(RuleSet ruleSet, string filePath)
             FilePath = filePath;
         }
 
-        public void Save()
+        public void Save(ISourceControlledFileSystem sourceControlledFileSystem)
         {
-            RuleSet.WriteToFile(FilePath);
+            sourceControlledFileSystem.QueueFileWrite(FilePath, () =>
```

Comment:
This now looks weird: the binder now has to know to queue the file operation, rather than just executed it directly. Also, it isn't clear what the relationship between the `sourceControlledFileSystem` and the `fileSystem` is, if any. I think there's a design issue here we need to clear up.